lib/iex/lib/iex.ex

```diff
@@ -47,12 +44,78 @@ defmodule IEx do
 
       set ERL_AFLAGS "-kernel shell_history enabled"
 
+  ## Expressions in IEx
+
+  As an interactive shell, IEx evaluates expressions. This has some
+  interesting consequences that are worth discussing.
+
+  The first one is that the code is truly evaluated and not compiled.
+  This means that any benchmarking done in the shell is going to have
+  skewed results. So never run any profiling nor benchmarks in the shell.
+
+  Second, IEx allows you to break an expression into many lines,
+  since this is common in Elixir. For example:
+
+      iex(1)> "ab
+      ...(1)> c"
+      "ab\nc"
+
+  In the example above, the shell will be expecting more input until it
+  finds the closing quote. Sometimes it is not obvious which character
+  the shell is expecting, and the user may find themselves trapped in
+  the state of incomplete expression with no ability to terminate it other
+  than by exiting the shell.
+
+  For such cases, there is a special break-trigger (`#iex:break`) that when
+  encountered on a line by itself will force the shell to break out of any
+  pending expression and return to its normal state:
+
+      iex(1)> ["ab
+      ...(1)> c"
+      ...(1)> "
+      ...(1)> ]
+      ...(1)> #iex:break
+      ** (TokenMissingError) iex:1: incomplete expression
```

Comment:
Given this section, should we talk somewhere about interrupt from the `Ctrl-g` menu? It's more powerful in a way it can solve this situation, but also infinite loops & waiting in a `receive` without a message.